lib/api/core.dart

```diff
@@ -80,6 +96,22 @@ class LiveApiConnection extends ApiConnection {
     }
     return utf8.decode(response.bodyBytes);
   }
+
+  @override
+  Future<String> postFileFromStream(String route, Stream<List<int>> content, int length, { String? filename }) async {
+    assert(_isOpen);
+    http.MultipartRequest request = http.MultipartRequest('POST', Uri.parse("${auth.realmUrl}/api/v1/$route"))
+      ..files.add(http.MultipartFile('file', content, length, filename: filename))
+      ..headers.addAll(_headers());
```

Comment:
I noticed that `zulip-mobile` was using the header `Content-Type: multipart/form-data` for all and only the requests to send a file to the server, so I figured we wanted that header too. This [`MultipartRequest`](https://pub.dev/documentation/http/latest/http/MultipartRequest-class.html) class seemed like a convenient way to make a request with that header—

> A `multipart/form-data` request.

—and then I was also happy to see the convenient interface for uploading files, with `..files.add`, etc.

For how to represent the file, I chose `Stream<List<int>> content` instead of a path, hoping that would push any file-access issues (like with permissions) out to the caller to deal with, and letting our `/api/` code continue to be just a channel for data going to and from the server.